packages/components/src/assay/AssayDesignPage.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { FC, memo, useCallback, useEffect, useState } from 'react';
+import { List, Map } from 'immutable';
+import { WithRouterProps } from 'react-router';
+
+import { InjectedAssayModel } from '../internal/components/assay/withAssayModels';
+import { InjectedRouteLeaveProps, useRouteLeave } from '../internal/util/RouteLeave';
+import { useNotificationsContext } from '../internal/components/notifications/NotificationsContext';
+import { AssayProtocolModel } from '../internal/components/domainproperties/assay/models';
+import { useServerContext } from '../internal/components/base/ServerContext';
+import { fetchProtocol } from '../internal/components/domainproperties/assay/actions';
+import { DomainDesign, IDomainField } from '../internal/components/domainproperties/models';
+import { GENERAL_ASSAY_PROVIDER_NAME } from '../internal/components/assay/constants';
+import { clearAssayDefinitionCache } from '../internal/components/assay/actions';
+
+import { AppURL } from '../internal/url/AppURL';
+import { ASSAYS_KEY } from '../internal/app/constants';
+import { hasPremiumModule, sampleManagerIsPrimaryApp } from '../internal/app/utils';
+import { InsufficientPermissionsPage } from '../internal/components/permissions/InsufficientPermissionsPage';
+import { LoadingPage } from '../internal/components/base/LoadingPage';
+import { Page } from '../internal/components/base/Page';
+
+import { Alert } from '../internal/components/base/Alert';
+import { getActionErrorMessage } from '../internal/util/messaging';
+import { AssayDesignerPanels } from '../internal/components/domainproperties/assay/AssayDesignerPanels';
+import { ProductMenuModel } from '../internal/components/navigation/model';
+
+import { DEFAULT_SAMPLE_FIELD_CONFIG } from '../internal/components/samples/constants';
+
+import { protocolHasSample, renderSampleRequiredPanelHeader } from './SampleRequiredDomainHeader';
+import { AssayHeader } from './AssayHeader';
+import { onAssayDesignChange } from './actions';
+import { useAssayAppContext } from './AssayAppContext';
+import { assayPage } from './AssayPageHOC';
+
+const ASSAY_DESIGNER_HEADER = 'Connect your experimental results to samples for rich data connections.';
+
+// Need to do an exclusion list to allow unknown file-based assays to have batch fields
+const REMOVE_BATCH_DOMAIN_ASSAYS = [
+    'General',
+    'ELISA',
+    'ELISpot',
+    'Noblis Simple',
+    'NAb',
+    'Viability',
+    'TZM-bl Neutralization (NAb)',
+    'TZM-bl Neutralization (NAb), High-throughput (Cross Plate Dilution)',
+    'TZM-bl Neutralization (NAb), High-throughput (Single Plate Dilution)',
+];
+
+interface OwnProps {
+    goBack: (n?: number) => void;
+    menu: ProductMenuModel;
+    menuInit: (invalidate?: boolean) => void;
+    navigate: (url: string | AppURL, replace?: boolean) => void;
```

Comment:
FYI, I did create a CommonPageProps interface for these three. This one can be updated to use it after things are merged together.

Reply:
Thanks. Updated to use that.